GVFS/GVFS/CommandLine/ConfigVerb.cs

```diff
@@ -31,30 +46,101 @@ protected override string VerbName
 
         public override void Execute()
         {
-            this.localConfig = new LocalGVFSConfig();
+            if (GVFSPlatform.Instance.IsUnderConstruction)
+            {
+                this.ReportErrorAndExit("`gvfs config` is not yet implemented on this operating system.");
+            }
 
+            this.localConfig = new LocalGVFSConfig();
             string error = null;
-            bool isRead = string.IsNullOrEmpty(this.Value);
 
-            if (isRead)
+            if (this.IsMutuallyExclusiveOptionsSet(out error))
             {
-                string value = null;
-                if (this.localConfig.TryGetConfig(this.Key, out value, out error, tracer: null))
+                this.ReportErrorAndExit(error);
+            }
+
+            if (this.List)
+            {
+                Dictionary<string, string> allSettings;
+                if (!this.localConfig.TryGetAllConfig(out allSettings, out error))
                 {
-                    Console.WriteLine(value);
+                    this.ReportErrorAndExit(error);
                 }
-                else
+
+                const string ConfigOutputFormat = "{0}={1}";
+                foreach (KeyValuePair<string, string> setting in allSettings)
                 {
-                    this.ReportErrorAndExit(error);
+                    Console.WriteLine(ConfigOutputFormat, setting.Key, setting.Value);
                 }
+
+                return;

```

Comment:
I would drop all of these return statements, because it's generally easier to follow a method with one exit point. In this case, you've already confirmed that each of the if's are mutually exclusive, but even so, you could further reinforce that with else statements. Either way there's no value in the returns and they just add mental overhead.

Reply:
replaced returns with if else